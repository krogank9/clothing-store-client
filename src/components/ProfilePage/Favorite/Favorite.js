import React from 'react';

import { Link } from "react-router-dom";
import Utils from "../../../utils";

class Favorite extends React.Component {

  getProductLink = () => {
    return `/products/${Utils.normalizeName(this.props.product.name)}.${this.props.product.id}`
  }

  getSize = () => {
    // todo implement sizes for cart, favorite, & product pages
    //this.props.size
    return false//"Medium"
  }

  getPrice = () => {
    return Utils.normalizePrice(this.props.product.price)
  }

  render() {
    return (
      <tr>
        <td>
          <div className="cart-item">
            <div className="cart-item-img-container">
              <Link to={this.getProductLink()}>
                <img className="cart-product-img" src={`${process.env.PUBLIC_URL}/assets/products/${Utils.normalizeName(this.props.product.name)}.jpg`}></img>
              </Link>
            </div>
            <div className="cart-item-info">
              <div><Link to={this.getProductLink()}>{this.props.product.name}</Link></div>
              <small>${this.getPrice()}</small>
              <br />
              <Link to="#" onClick={this.props.onRemoveClicked}>Remove Favorite</Link>
            </div>
          </div>
        </td>
      </tr>
    );
  }
}

export default Favorite;
