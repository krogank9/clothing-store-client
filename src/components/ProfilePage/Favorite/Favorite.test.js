import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from "react-router-dom";
import Favorite from './Favorite';

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe(`Favorite component`, () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <table><tbody>
          <Favorite />
        </tbody></table>
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });


  it('renders Favorite', () => {
    const wrapper = shallow(
      <BrowserRouter>
        <table><tbody>
          <Favorite />
        </tbody></table>
      </BrowserRouter>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})