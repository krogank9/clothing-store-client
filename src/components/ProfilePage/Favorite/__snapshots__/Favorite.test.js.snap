// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite component renders Favorite 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <table>
    <tbody>
      <Favorite
        product={Object {}}
      />
    </tbody>
  </table>
</Router>
`;
